### ISSUE TITLE
Leiningen 2.8.0 requires Java 1.8.0u131 by default

### ISSUE THREAD

**Issue by brixtonasias** [technomancy/leiningen#2321]
When trying to run a build with a Java version lower than `1.8.0u131` builds fail with the message: `Unrecognized VM option 'UseCGroupMemoryLimitForHeap'`

The website tells users that: 

> Make sure you have a Java JDK version 6 or later.

The fix in the FAQ to use `:jvm-opts ^:replace []` in `project.clj` results in:

```
Exception in thread "main" java.lang.RuntimeException: No reader function for tag object, compiling:(/private/var/folders/hl/wc_8mc8950dbbp5pdnz56br00000gn/T/form-init1787191864903406657.clj:1:8328)
```

What is the actual solution to this when one can not / does not want to upgrade the Java version?

**Comment by acron0** [comment]
```clojure
(def ^:private cgroups-jvm-opts
  ;; this assumes the JVM version Leiningen is run under matches the project
  (let [[v u] (re-seq #"[^-_]+" (System/getProperty "java.runtime.version"))]
    (if (or (and (= v "1.8.0") (>= (Integer. u) 131))
            (not= v "1.7.0")
            (not= v "1.6.0"))
      ["-XX:+UnlockExperimentalVMOptions" "-XX:+UseCGroupMemoryLimitForHeap"])))
```
This code is wrong. When `v=1.8.0` and `u<131` the `or` will return `true` due to the `not=` clauses.

```clojure
user> (let [[v u] ["1.8.0" "111"]]
        (or (and (= v "1.8.0") (>= (Integer. "111") 131))
            (not= v "1.7.0")
            (not= v "1.6.0")))
true
```
I'm curious as to the `not=` clauses and why they were included.

**Comment by technomancy** [comment]
It looks like this is really two bugs; one detecting the JVM version and one preventing the manual override from working. 

Can you gist the contents of `/private/var/folders/hl/wc_8mc8950dbbp5pdnz56br00000gn/T/form-init1787191864903406657.clj` so we can address the latter?

Also a great reminder to update the web site to recommend a version of the JDK that still gets security updates.

**Comment by scstarkey** [comment]
I have it even worse. I'm using the latest Java 8. Here's my environment (with no changes to `project.clj`):

`$ lein version`
`Leiningen 2.8.0 on Java 1.8.0_152 Java HotSpot(TM) 64-Bit Server VM`

If I use `lein run` (which runs jetty9 manually) my application runs fine. If I use `lein ring server` I get the same error as the OP:

`$ lein ring server`
`Java HotSpot(TM) 64-Bit Server VM warning: Unable to open cgroup memory limit file /sys/fs/cgroup/memory/memory.limit_in_bytes (No such file or directory)
Exception in thread "main" java.lang.RuntimeException: No reader function for tag object, compiling:(/private/var/folders/9d/mkhl6xx16mb983xg01lnshv80000gn/T/form-init8708870852094912139.clj:1:8275)`
Sadly the file is temporary so I can't get the contents.

If I go back to `leiningen 2.7.0` I experience no issues.

**Comment by scstarkey** [comment]
Just tried in JDK 9.0.1 and the same issue occurs.

**Comment by technomancy** [comment]
@scstarkey can you run the command with `:eval-in :pprint` in `project.clj`? Also, do you have multiple JVMs installed?

**Comment by scstarkey** [comment]
Here's a bunch of stuff @technomancy --https://gist.github.com/scstarkey/32d999be77ceca2e5759a303d8857d75

Yes.. I have a few different JDKs.

**Comment by technomancy** [comment]
Oh, I see; this is the same unreadable form problem as above. It's unrelated to the question of which Java version is being used, but it can be triggered by the workaround for the version problem. Thanks.

**Comment by scstarkey** [comment]
Happy to help!

**Comment by technomancy** [comment]
Tracking the unreadable form issue separately in #2328.

**Comment by technomancy** [comment]
Fixed in #2324.
